error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/Slf4jLogDeclaration.java

```diff
@@ -0,0 +1,136 @@
+import static com.google.errorprone.BugPattern.SeverityLevel.WARNING;
+import static com.google.errorprone.BugPattern.StandardTags.LIKELY_ERROR;
+import static com.google.errorprone.matchers.Matchers.isSubtypeOf;
+import static com.google.errorprone.matchers.Matchers.staticMethod;
+import static tech.picnic.errorprone.bugpatterns.util.Documentation.BUG_PATTERNS_BASE_URL;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.ErrorProneFlags;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.ClassTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.Tree.Kind;
+import com.sun.source.tree.VariableTree;
+import com.sun.source.util.TreeScanner;
+import javax.inject.Inject;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.Name;
+import javax.tools.JavaFileObject;
+import org.jspecify.annotations.Nullable;
+import tech.picnic.errorprone.bugpatterns.util.SourceCode;
+
+/**
+ * A {@link BugChecker} that warns when SLF4J declarations are not canonicalized across the project.
+ *
+ * @apiNote The default canonicalized logger name can be overriden through {@link ErrorProneFlags
+ *     flag arguments}.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary = "Make sure SLF4J log declarations follow the SLF4J conventions inside classes.",
+    link = BUG_PATTERNS_BASE_URL + "Slf4jLogDeclaration",
+    linkType = CUSTOM,
+    severity = WARNING,
+    tags = LIKELY_ERROR)
+public final class Slf4jLogDeclaration extends BugChecker implements ClassTreeMatcher {
+  private static final long serialVersionUID = 1L;
+  private static final Matcher<Tree> LOGGER = isSubtypeOf("org.slf4j.Logger");
+  private static final String CANONICALIZED_LOGGER_NAME_FLAG =
+      "Slf4jLogDeclaration:CanonicalizedLoggerName";
+  private static final String DEFAULT_CANONICALIZED_LOGGER_NAME = "LOG";
+  private static final Matcher<ExpressionTree> GET_LOGGER_METHOD =
+      staticMethod().onDescendantOf("org.slf4j.LoggerFactory").named("getLogger");
+
+  private final String canonicalizedLoggerName;
+
+  /** Instantiates a default {@link Slf4jLogDeclaration} instance. */
+  public Slf4jLogDeclaration() {
+    this(ErrorProneFlags.empty());
+  }
+
+  /**
+   * Instantiates a customized {@link Slf4jLogDeclaration}.
+   *
+   * @param flags Any provided command line flags.
+   */
+  @Inject
+  Slf4jLogDeclaration(ErrorProneFlags flags) {
+    canonicalizedLoggerName = getCanonicalizedLoggerName(flags);
```

Comment:
We should add 1 test case where we pass a different logger name to this BugPattern and validate that it is correct. 

See this code from `LexicographicalAnnotationAttributeListingTest`:
```
 CompilationTestHelper.newInstance(LexicographicalAnnotationAttributeListing.class, getClass())
        .setArgs(
            ImmutableList.of(
                "-XepOpt:LexicographicalAnnotationAttributeListing:Includes=pkg.A.Foo,pkg.A.Bar",
                "-XepOpt:LexicographicalAnnotationAttributeListing:Excludes=pkg.A.Bar#value"))
```
We can do the same but for our flags here :).